test/e2e/registration_service_test.go

```diff
@@ -429,6 +431,177 @@ func (s *registrationServiceTestSuite) TestSignupOK() {
 	assert.Equal(s.T(), ExpectedConsoleURL(s.T(), s.memberAwait, memberCluster), mp["consoleURL"])
 }
 
+func (s *registrationServiceTestSuite) TestPhoneVerification() {
```

Comment:
The test has 170 lines... This is a pain to follow/understand - I got to the middle of the test and lost the whole context. Could you please break it down to either multiple subtests or multiple helper functions?

Reply:
It was a monumental effort to actually get the entire phone verification workflow tested and working successfully so yes, the size of the test did blow out a bit - I totally agree though that it's a bit too large to follow, so I'll make some changes to turn some of the duplicated code into reusable functions.

Reply:
I've refactored it down to 100 lines, let me know if that looks ok now.